Oleksandr Khomin/contracts/Token.sol

```diff
@@ -0,0 +1,86 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+contract Proxy {
+    address public implementation;
+    address public owner;
+
+    constructor(address _implementation) {
+        implementation = _implementation;
+        owner = msg.sender;
+    }
+
+    function setImplementation(address _implementation) external {
+        require(msg.sender == owner);
+        implementation = _implementation;
+    }
+
+    function _delegate(address _imp) internal virtual {
+        assembly {
+            calldatacopy(0, 0, calldatasize())
+
+            let result := delegatecall(gas(), _imp, 0, calldatasize(), 0, 0)
+
+            returndatacopy(0, 0, returndatasize())
+
+            switch result
+            case 0 {
+                revert(0, returndatasize())
+            }
+            default {
+                return (0, returndatasize())
+            }
+        }
+    }
+
+    fallback() external payable {
+        _delegate(implementation);
+    }
+}
+
+contract TokenV1 {
+    string public name = "Khomin Token";
+    string public symbol = "KHT";
+
+    // BUG: should be uint256
+    int256 public totalSupply = 1000;
```

Comment:
but here actually 2 bugs (not one). First: int256 (and no check `require(balances[msg.sender] >= 0);`), second - no `require(balances[msg.sender] >= amount);`